pkg/pool-linear/test/GearboxLinearPool.test.ts

```diff
@@ -0,0 +1,117 @@
+
+  const POOL_SWAP_FEE_PERCENTAGE = fp(0.01);
+
+  before('setup', async () => {
+    [, lp, trader, owner] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy tokens', async () => {
+    mainToken = await Token.create('USDC');
+    gearboxVault = await deploy('MockGearboxVault', { args: [mainToken.address] });
+    wrappedTokenInstance = await deploy('MockGearboxDieselToken', {
+      args: ['dUSDC', 'dUSDC', 6, gearboxVault.address],
+    });
+    wrappedToken = await Token.deployedAt(wrappedTokenInstance.address);
+
+    tokens = new TokenList([mainToken, wrappedToken]).sort();
+
+    await tokens.mint({ to: [lp, trader], amount: fp(100) });
+  });
+
+  sharedBeforeEach('deploy pool factory', async () => {
+    vault = await Vault.create();
+    const queries = await deploy('v2-standalone-utils/BalancerQueries', { args: [vault.address] });
+    poolFactory = await deploy('GearboxLinearPoolFactory', {
+      args: [vault.address, vault.getFeesProvider().address, queries.address],
+    });
+  });
+
+  sharedBeforeEach('deploy and initialize pool', async () => {
+    const tx = await poolFactory.create(
+      'Balancer Pool Token',
+      'BPT',
+      mainToken.address,
+      wrappedToken.address,
+      bn(0),
+      POOL_SWAP_FEE_PERCENTAGE,
+      owner.address
+    );
+
+    const receipt = await tx.wait();
+    const event = expectEvent.inReceipt(receipt, 'PoolCreated');
+
+    pool = await LinearPool.deployedAt(event.args.pool);
+  });
+
+  describe('asset managers', () => {
+    it('sets the same asset manager for main and wrapped token', async () => {
+      const poolId = await pool.getPoolId();
+
+      const { assetManager: firstAssetManager } = await vault.getPoolTokenInfo(poolId, tokens.first);
+      const { assetManager: secondAssetManager } = await vault.getPoolTokenInfo(poolId, tokens.second);
+
+      expect(firstAssetManager).to.equal(secondAssetManager);
+    });
+
+    it('sets the no asset manager for the BPT', async () => {
+      const poolId = await pool.getPoolId();
+      const { assetManager } = await vault.instance.getPoolTokenInfo(poolId, pool.address);
+      expect(assetManager).to.equal(ZERO_ADDRESS);
+    });
+  });
+
+  describe('getWrappedTokenRate', () => {
+    it('returns the expected value', async () => {
+      // Reserve's normalised income is stored with 27 decimals (i.e. a 'ray' value)
+      // 1e27 implies a 1:1 exchange rate between main and wrapped token
+      await gearboxVault.setRate(bn(1e27));
+      expect(await pool.getWrappedTokenRate()).to.be.eq(fp(1));
+
+      // We now double the reserve's normalised income to change the exchange rate to 2:1
+      await gearboxVault.setRate(bn(2e27));
+      expect(await pool.getWrappedTokenRate()).to.be.eq(fp(2));
+    });
+  });
+
+  describe('constructor', () => {
+    it('reverts if the mainToken is not the ASSET of the wrappedToken', async () => {
+      const otherToken = await Token.create('USDC');
```

Comment:
I would change this to DAI just to make clear that it's a different token.
```suggestion
      const otherToken = await Token.create('DAI');
```